x-pack/plugins/fleet/server/services/epm/archive/parse.ts

```diff
@@ -512,3 +476,56 @@ export function parseAndVerifyInputs(manifestInputs: any, location: string): Reg
   }
   return inputs;
 }
+
+export function parseDataStreamElasticsearchEntry(
+  elasticsearch: Record<string, any>,
+  ingestPipeline?: string
+) {
+  const parsedElasticsearchEntry: Record<string, any> = {};
+
+  if (ingestPipeline) {
+    parsedElasticsearchEntry['ingest_pipeline.name'] = ingestPipeline;
+  }
+
+  if (elasticsearch?.privileges) {
+    parsedElasticsearchEntry.privileges = elasticsearch.privileges;
+  }
+
+  if (elasticsearch?.source_mode) {
```

Comment:
Source mode can only be one of two values, not sure if we should be validating here? We only seem to do high level validation